android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt

```diff
@@ -79,11 +97,14 @@ class MobileWalletAdapter(
                     val authResult = credsState.let { creds ->
                         if (creds is CredentialState.Provided) {
                             with (creds.credentials) {
-                                if (authToken == null) {
+                                val authResult = authToken?.let { token ->
+                                    adapterOperations.reauthorize(identityUri, iconUri, identityName, token)
+                                } ?: run {
                                     adapterOperations.authorize(identityUri, iconUri, identityName, rpcCluster)
-                                } else {
-                                    adapterOperations.reauthorize(identityUri, iconUri, identityName, authToken)
                                 }
+
+                                authToken = authResult.authToken
+                                authResult
                             }
                         } else {
```

Comment:
should an error be returned here? or is the thought that calling privileged methods in block will trigger the error if no credentials have been provided?

Reply:
Returning null is the fallback to keep using it the "old" way of manually doing auth/reauth yourself.

Reply:
discussed offline, no error is returned so that existing implementations will continue to work as is.